packages/machine-extractor/src/extractAction.ts

```
@@ -0,0 +1,411 @@
+          t.isNumericLiteral(foundDelay.value)
+        ) {
+          delay = {
+            type: getLiteralType(foundDelay.value),
+            value: foundDelay.value.value,
+          };
+        }
+      }
+
+      // id = string
+      const foundId = arg3.properties.find(
+        prop =>
+          t.isObjectProperty(prop) &&
+          t.isIdentifier(prop.key) &&
+          prop.key.name === 'id',
+      ) as t.ObjectProperty | undefined;
+      if (
+        foundId &&
+        (t.isStringLiteral(foundId.value) || t.isNumericLiteral(foundId.value))
+      ) {
+        id = {
+          type: getLiteralType(foundId.value),
+          value: foundId.value.value,
+        };
+      }
+    }
+
+    return {
+      event: eventObject,
+      actorRef: actorRef!,
+      options: {
+        delay,
+        id,
+      },
+    };
+  }
+
+  throw Error(`Unsupported sendTo expression`);
+}
+
+function getLiteralType(value: t.ObjectProperty['value']) {
+  if (
+    t.isNullLiteral(value) ||
+    t.isRegExpLiteral(value) ||
+    t.isTemplateLiteral(value)
+  ) {
+    throw Error('null,regexp or template literals can not have literal type');
+  }
+  if (t.isNumericLiteral(value)) {
+    return 'number';
+  }
+  if (t.isStringLiteral(value)) {
+    return 'string';
+  }
+  if (t.isBooleanLiteral(value)) {
+    return 'boolean';
+  }
+  throw Error('Unsupported literal property value');
+}
+
+export const isBuiltinActionWithName = (actionNode: ActionNode, name: string) =>
+  t.isCallExpression(actionNode.node) &&
+  t.isIdentifier(actionNode.node.callee) &&
+  actionNode.node.callee.name === name;
+
+function extractEventObject(
+  eventObject: t.ObjectExpression,
+  fileContent: string,
+): Record<string, { type: JsonEntryType; value: JsonValue }> {
+  const extracted: Record<
+    string,
+    { type: JsonEntryType; value: JsonValue }
+  > = {};
+  eventObject.properties.forEach(prop => {
+    if (t.isObjectProperty(prop)) {
+      if (t.isIdentifier(prop.key)) {
```

You should throw/warn/do something about unrecognized patterns - fail fast instead of propagating incomplete/broken results forward.